src/Shape.cpp

```diff
@@ -10,13 +10,31 @@
 
 namespace geo {
 
+class Line
+{
+public:
+    Line(const geo::Vector3& a, const geo::Vector3& b) : a_(a), b_(b)
+    {
+        U_ = b_-a_;
+        V_ = a_.cross(b_);
+    }
+
+    inline const geo::Vector3& a() const { return a_; }
+    inline const geo::Vector3& b() const { return b_; }
+    inline const geo::Vector3& U() const { return U_; }
+    inline const geo::Vector3& V() const { return V_; }
+
+protected:
+    const geo::Vector3& a_, b_;
+    geo::Vector3 U_, V_;
```

Comment:
Add a description of these variables and give them more meaningful names.

Reply:
I think I keep these names, as these match the 'paper'. But I will add descriptions and a link.